### PR TITLE
[Visual Script]: make into new package and rename nodes for new groups

### LINKED ISSUE #9751: Rename Behave Graph to Visual Script
make sure attribution license is still correct

### PR DESCRIPTION
**Pull request by SYBIOTE**
seperates visual script into its own package with the relevant UI seperate from the editor and engine packages

### CODE REVIEW

File: packages/visual-script/src/engine/generateUuid.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  const d1 = (Math.random() * 0xffffffff) | 0
+  const d2 = (Math.random() * 0xffffffff) | 0
+  const d3 = (Math.random() * 0xffffffff) | 0
+  const uuid = `${lut[d0 & 0xff] + lut[(d0 >> 8) & 0xff] + lut[(d0 >> 16) & 0xff] + lut[(d0 >> 24) & 0xff]}-${
+    lut[d1 & 0xff]
+  }${lut[(d1 >> 8) & 0xff]}-${lut[((d1 >> 16) & 0x0f) | 0x40]}${lut[(d1 >> 24) & 0xff]}-${lut[(d2 & 0x3f) | 0x80]}${
+    lut[(d2 >> 8) & 0xff]
+  }-${lut[(d2 >> 16) & 0xff]}${lut[(d2 >> 24) & 0xff]}${lut[d3 & 0xff]}${lut[(d3 >> 8) & 0xff]}${
+    lut[(d3 >> 16) & 0xff]
+  }${lut[(d3 >> 24) & 0xff]}`
```
[the remark this answers is not in the document]
Comment (DanielBelmes) [reply packages/visual-script/src/engine/generateUuid.ts]: @SYBIOTE if we're gonna be generating a uuid might as well use this standard libbrary https://github.com/uuidjs/uuid

[the remark this answers is not in the document]
Comment (HexaField) [reply packages/visual-script/src/engine/generateUuid.ts]: we have uuidv4 and also threejs utility, we should standardize on one of them

[the remark this answers is not in the document]
Comment (DanielBelmes) [reply packages/visual-script/src/engine/generateUuid.ts]: > we have uuidv4 and also threejs utility, we should standardize on one of them

I believe the one I linked is the same as uuidv4. I like that one. It's super standard and will definitly be more secure than three's as it has more users.

**Review by DanielBelmes — CHANGES_REQUESTED** [review]
Just wanted to bring to your attention the failing checks and merge conflicts

**Comment by SYBIOTE** [comment]
will handle these, the visual script -tests must be merged into this branch first

**Comment by DanielBelmes** [comment]
> will handle these, the visual script -tests must be merged into this branch first

Ah understood. I will check out that PR

**Comment by SYBIOTE** [comment]
its all fixed

**Pull request merged**